ci: GitHub actions refactor

notpeter opened the pull request:
Refactor GitHub actions CI workflow.
- Single combined 'tests_pass' action so we only need one mandatory check for merge queue 
- Add new `job_spec` job which determines what needs to be run (+5secs)
  - Do not run full CI for docs only changes (~30secs vs 10+mins)
  - Only run `script/generate-licenses` if Cargo.lock changed (saves ~23secs on mac_test)
- Move prettier /docs check to ci.yml and remove docs.yml 
- Run Windows tests on every PR commit
- Added new Windows runners named to reflect their OS/capacity (windows-2025-64, windows-2025-32,  windows-2025-16)

Release Notes:

- N/A

maxdeviant commented on .github/workflows/ci.yml:
```diff
@@ -372,13 +414,42 @@ jobs:
+    name: Tests Pass
+    runs-on: ubuntu-latest
+    needs:
+      - style
```
I think we should always run `style`, as it contains things like our typos checks and checks for `todo!` comments.

notpeter committed:
ci: GitHub actions refactor

notpeter committed:
More improvements.
Remove temporary windows runner data capture.
Always require style checks.
Revert larger Linux runner.

notpeter committed:
Corrections.

notpeter committed:
Fetch less git for job_spec.

notpeter committed:
Merge remote-tracking branch 'origin/main' into refactor_gha

notpeter committed:
Update comment

notpeter committed:
Support non-main PR targets

notpeter committed:
Move prettier out of docs.yml

notpeter committed:
Add missing pnpm/action-setup

notpeter committed:
Always run style

notpeter committed:
Only run script/generate-licenses if Cargo.lock/Cargo.toml changed.

notpeter committed:
Merge branch 'main' into refactor_gha

notpeter committed:
Full Cargo.lock

notpeter commented on .github/workflows/ci.yml:
```diff
@@ -23,9 +23,43 @@ env:
+          else
+            echo "run_tests=false" >> $GITHUB_OUTPUT
+          fi
+          if [[ $(git diff --name-only $MERGE_BASE ${{ github.sha }} | grep '^Cargo.lo') ]]; then
```
Was not intentional. Nice catch.

I started to write `.lock` and then thought for a minute about whether I needed to `grep -E 'Cargo\.(lock|toml)'` and decided that was overkill as `Cargo.lock` should be sufficient.
